python/taichi/types/ndarray_type.py

```diff
@@ -25,6 +30,12 @@ def __init__(self,
                  element_dim=None,
                  element_shape=None,
                  field_dim=None):
+        # The element shape are deprecated. Use dtype to manage element-wise arguments.
+        if element_dim is not None or element_shape is not None:
+            warnings.warn(
```

Comment:
nit: we'd better say explicitly when it'll be removed and add a test in `test_deprecations.py` to track ;)

Reply:
Actually shall we remove changes in this file to the next PR? 
Since changes in annotations can go in immediately, but ideally there should be a separate PR with that updates the logic here. For example, after throwing the warning we should just translate `element_shape` and `element_dim` into corresonding `dtype`s and update the check logic to handle `dtype` only. 
Then it's much easier to maintain and delete deprecated `element_dim/element_shape` in the future. ;)

Reply:
Great, then I should move this warning in the next PR that removes internal use of element shape!